app/code/Dariam/ControllerDemos/Controller/Demos/RawResponseDemo.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types=1);
+
+namespace Dariam\ControllerDemos\Controller\Demos;
+
+use Magento\Framework\Controller\Result\Raw;
+
+class RawResponseDemo implements
+    \Magento\Framework\App\Action\HttpGetActionInterface
+{
+    private \Magento\Framework\Controller\Result\RawFactory $rawFactory;
+
+    /**
+     * @param \Magento\Framework\Controller\Result\RawFactory $rawFactory
+     */
+    public function __construct(
+        \Magento\Framework\Controller\Result\RawFactory $rawFactory
+    ) {
+        $this->rawFactory = $rawFactory;
+    }
+
+    /**
+     * @return Raw
+     */
+    public function execute(): Raw
+    {
+        $result = $this->rawFactory->create();
+
+        return $result->setHeader('Content-Type', 'text/html')
+            ->setContents(<<<HTML
+<ul>
+    <li><a href="https://daria-malenko-magento.local/dariam-controller-demos/demos/redirectresponsedemo">RedirectResponseDemo</a></li>
```

Comment:
@dariama3 
1) Redirect leads to another domain. Thus, a link must open in a new tab.
2) Links must not contain domain to work with current domain.